test/Numerics/LinearSolvers/cg.jl

```diff
@@ -48,18 +49,22 @@ let
         @test norm(A * x - b) / norm(b) < err_thresh
     end
 
-    at_A = ArrayType(A)
-    at_b = ArrayType(b)
-    at_x = ArrayType(ones(n) * 1.0)
-    mulbyat_A!(y, x) = (y .= at_A * x)
-    at_method(b, tol) = ConjugateGradient(b, max_iter = n)
-    linearsolver = at_method(at_b, tol)
-    iters = linearsolve!(mulbyat_A!, linearsolver, at_x, at_b; max_iters = n)
-    exact = at_A \ at_b
-
-    @testset "Architecture-specific Array test" begin
-        @test norm(at_x - exact) / norm(exact) < err_thresh
-        @test norm(at_A * at_x - at_b) / norm(at_b) < err_thresh
+    # Testing for CuArrays
+    if CUDAapi.has_cuda_gpu()
+        at_A = ArrayType(A)
+        at_b = ArrayType(b)
+        at_x = ArrayType(ones(n) * 1.0)
```

Comment:
Should we type-cast `FT` here?

Reply:
It might be nice to extend these tests to `Float32`

Reply:
yeah, I can do that. I wanted to loop over all arraytypes as well. I'll make it a bit more concise.